rust/candid/src/parser/typing.rs

```diff
@@ -261,6 +264,90 @@ fn check_actor(env: &Env, actor: &Option<IDLType>) -> Result<Option<Type>> {
     }
 }
 
+pub(crate) fn interpret_number(number: IDLValue, typ: IDLType) -> Result<IDLValue> {
```

Comment:
There is a `annotate_type` function in `value.rs`. It's better to reuse that function. We can convert the `IDLType` to `Type` by calling `typing::check_type`. This way, we can give arbitrary type annotations, not just restricting to numbers.

Reply:
Oh wow.. I wasn't aware of that function. Thanks! Indeed it makes sense to use that one. Will update.

Reply:
One problem is these functions need an environment (`typing::Env` for `chekc_type`, `TyEnv` for `annotate_type`) which we don't have during parsing. Should I create an empty environment? Doesn't make sense to me and the error messages will probably suffer in some edge cases, but perhaps that's fine for now.

Reply:
An empty environment is probably the right thing (for now). These annotations in textual values, by construction, cannot refer to named types (there is no place to name them – yet).

Reply:
OK, just did that. There's a new problem though which is `annotate_type` doesn't allow annotating e.g. "123" with float32, which interpret_number allowed. So we again don't have a syntax for floats.

Should `annotate_type` allow annotating a number with a float type?

Reply:
`123.` represents a float. I think this is consistent with Motoko.